src/tools/perf/lib/ucp_tests.cc

```diff
@@ -310,6 +353,11 @@ class ucp_perf_test_runner {
         }
     }
 
+    void fence()
+    {
+        ucp_worker_fence(m_perf.ucp.worker);
```

Comment:
maybe move this inside send_last_iter (only if UCX_PERF_CMD_PUT is used)

Reply:
done